File: go/clock/clock.go
// Package clock contains methods for clock type creation/manipulation.
package clock

import "fmt"

const minutesPerDay = 1440;

// Clock type holds the time of day.
type Clock int

// New creates a new clock with the given time.
func New(hours, minutes int) Clock {
	minutes = (hours*60 + minutes) % minutesPerDay

	for minutes < 0 {
		minutes += minutesPerDay
	}

	for minutes > minutesPerDay {
		minutes -= minutesPerDay
	}

	return Clock(minutes)
}

// String returns a string representatoin of the clock.
func (clock Clock) String() string {
	return fmt.Sprintf("%02v:%02v", int(clock)/60, int(clock)%60)
}

// Add increases the current time of the clock by the number of minutes.
func (clock Clock) Add(minutes int) Clock {
	return New(0, int(clock)+minutes)
}

// Subtract reduces the current time of the clock by the number of minutes.
func (clock Clock) Subtract(minutes int) Clock {
	return clock.Add(-minutes)
}

File: cpp/clock/clock.cpp
#include "clock.h"

namespace date_independent
{

    clock::clock(int mins)
    {
        mins %= clock::MINUTES_PER_DAY;
        mins += clock::MINUTES_PER_DAY;
        mins %= clock::MINUTES_PER_DAY;

        this->minutes = mins;
    }

    clock clock::at(int hours, int mins)
    {
        return clock((hours * 60) + mins);
    }

    clock clock::plus(int mins)
    {
        return this->at(0, this->minutes + mins);
    }

    clock clock::minus(int mins)
    {
        return this->plus(-mins);
    }

    bool operator==(const clock &lhs, const clock &rhs)
    {
        return lhs.minutes == rhs.minutes;
    }

    bool operator!=(const clock &lhs, const clock &rhs)
    {
        return !(lhs == rhs);
    }

    clock::operator std::string() const
    {
        char buffer[6];
        sprintf(buffer, "%02d:%02d", this->minutes / 60, this->minutes % 60);

        return buffer;
    }

}  // namespace date_independent
